p5/core/vertex.py

```diff
@@ -83,13 +88,15 @@ def curve_vertex(x, y, z=0):
 
     if p5.mode == "3D":
         return
-    else:
-        if is_contour:
+    if builtins.current_renderer == 'vispy':
+        if has_contour:
```

Comment:
I think this should be `is_contour` instead. If an `end_contour` has been called, we probably shouldn't add more vertices to `curr_contour`.

Reply:
(this comment applies to other places in this file as well)

Reply:
Hi Mark. There was one more thing to this. Once we call a `begin_contour`  and end it with `end_contour` we don't reset the `is_contour` back to `False`. Apparently in p5.js something like 

```
beginShape()
vertex()
...
beginContour()
vertex()
...
endContour()
# vertex called again not in contour
vertex()
...
```
The vertex calls after `end_countour` are just neglected because the `is_contour` flag has been set are added to `contour_vertices`. I am unsure if this is a bug or `beginShape` is intended to be called in a way such that we first draw the shape followed by the contour and then end the shape, I didn't find anything in the docs.

However, to keep our output consistent with p5.js, we would also leave the `is_contour` variable to `True`.

I am now using `in_contour` as a check to see if we are inside a contour or not for the assert.

Reply:
I have added a comment in the docs for future reference.

Reply:
I see, that's a great observation! Thanks for bringing it up!